src/Logging/StdOutLogger.php

```diff
@@ -0,0 +1,144 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth\Logging;
+
+use InvalidArgumentException;
+use Psr\Log\LoggerInterface;
+use Psr\Log\LogLevel;
+use Stringable;
+
+/**
+ * A basic logger class to log into stdOut for GCP logging
+ */
+class StdOutLogger implements LoggerInterface
+{
+    /**
+     * @var array<string,int>
+     */
+    private array $levelMapping = [
+        LogLevel::EMERGENCY => 7,
+        LogLevel::ALERT => 6,
+        LogLevel::CRITICAL => 5,
+        LogLevel::ERROR => 4,
+        LogLevel::WARNING => 3,
+        LogLevel::NOTICE => 2,
+        LogLevel::INFO => 1,
+        LogLevel::DEBUG => 0,
+    ];
+    private int $level;
+
+    /**
+     * Constructs a basic PSR-3 logger class that logs into StdOut for GCP Logging
+     *
+     * @param string $level The level of the logger instance.
+     */
+    public function __construct(string $level = LogLevel::DEBUG)
+    {
+        $this->level = $this->getLevelMap($level);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function emergency(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::EMERGENCY, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function alert(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::ALERT, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function critical(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::CRITICAL, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function error(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::ERROR, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function warning(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::WARNING, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function notice(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::NOTICE, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function info(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::INFO, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function debug(string|Stringable $message, array $context = []): void
+    {
+        $this->log(LogLevel::DEBUG, $message);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function log($level, string|Stringable $message, array $context = []): void
+    {
+        if ($this->getLevelMap($level) < $this->level) {
+            return;
+        }
+
+        print($message . "\n");
+    }
+
+    /**
+     * @param string $levelName
+     * @return int
+     * @throws InvalidArgumentException
+     */
+    private function getLevelMap(string $levelName): int
```

Comment:
nit: to me `getLevelMap` would return the `levelMapping` variable. What if we instead called it `getLevelNumber` or `getLevelFromName`?